torch/_inductor/post_grad_custom_pass.py

```diff
@@ -0,0 +1,53 @@
+import hashlib
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import Tuple, Union
+
+import torch.fx.graph
+
+
+class PostGradCustomPass(ABC):
+    """
+    Implement this interface for post grad custom passes:
+
+    1) The __call__() method contains the implementation of the custom pass.
+
+    2) The uuid() method is necessary to support inductor caching compiled graphs when
+    your custom passes are applied. This method can return any unique identifier, as
+    long as it uniquely identifies your custom pass implementation. The caching logic
+    includes the ID in its key calculation, i.e., a new value will effectively
+    invalidate any existing entries. Most custom passes should depend purely on the
+    textual reprensentation of the code. In that case, we recommend using the
+    'get_hash_for_files` helper to compute a unique string based on the contents of a
+    list of source files, i.e., the source containing the custom pass
+    implementation. That ensures that any update to the code will map to new cache
+    entries. If your custom pass behaves differently based on some external state, then
+    you'll need to implement something more complicated (or disable caching).
+    """
+
+    @abstractmethod
+    def __call__(self, graph: torch.fx.graph.Graph) -> None:
+        """
+        Implementation of the custom pass.
+        """
+
+    @abstractmethod
+    def uuid(self) -> Union[bytes, str]:
+        """
+        Return an ID to uniquely identify your custom pass implementation. Return None
+        to bypass inductor code caching.
```

Comment:
nit: not sure if bypass means cache miss, or bypass means assume the cache is valid.

Reply:
bypass means cache miss, bypass the cache all together